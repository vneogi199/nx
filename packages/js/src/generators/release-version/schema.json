{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxJSReleaseVersionGenerator",
  "cli": "nx",
  "title": "Implementation details of `nx release version`",
  "description": "DO NOT INVOKE DIRECTLY WITH `nx generate`. Use `nx release version` instead.",
  "type": "object",
  "properties": {
    "projects": {
      "type": "array",
      "description": "The ProjectGraphProjectNodes being versioned in the current execution.",
      "items": {
        "type": "object"
      }
    },
    "projectGraph": {
      "type": "object",
      "description": "ProjectGraph instance"
    },
    "specifier": {
      "type": "string",
      "description": "Exact version or semver keyword to apply to the selected release group. NOTE: This should be set on the release group level, not the project level."
    },
    "preid": {
      "type": "string",
      "description": "The optional prerelease identifier to apply to the version, in the case that specifier has been set to prerelease."
    },
    "packageRoot": {
      "type": "string",
      "description": "The root directory of the directory (containing a manifest file at its root) to publish. Defaults to the project root"
    },
    "currentVersionResolver": {
      "type": "string",
      "default": "disk",
      "description": "Which approach to use to determine the current version of the project.",
      "enum": ["registry", "disk"]
    },
    "currentVersionResolverMetadata": {
      "type": "object",
      "description": "Additional metadata to pass to the current version resolver.",
      "default": {}
    }
  },
  "required": ["projects", "projectGraph", "specifier"]
}
