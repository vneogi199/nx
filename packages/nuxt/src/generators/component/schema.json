{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "NxNuxtComponent",
  "title": "Create a Nuxt Component",
  "description": "Create a Nuxt Component for Nx.",
  "type": "object",
  "examples": [
    {
      "command": "nx g component my-component --project=myapp",
      "description": "Generate a component in the `myapp` application"
    },
    {
      "command": "nx g component my-component --project=myapp --classComponent",
      "description": "Generate a class component in the `myapp` application"
    }
  ],
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this component?",
      "x-priority": "important"
    },
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the component?",
      "x-priority": "important"
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create `spec.ts` test files for the new component.",
      "default": false,
      "x-priority": "internal"
    },
    "flat": {
      "type": "boolean",
      "description": "Create component at the source root rather than its own directory.",
      "default": false
    },
    "directory": {
      "type": "string",
      "description": "Create the component under this directory (can be nested).",
      "alias": "dir",
      "x-priority": "important"
    },
    "pascalCaseFiles": {
      "type": "boolean",
      "description": "Use pascal case component file name (e.g. `App.vue`).",
      "alias": "P",
      "default": false
    },
    "pascalCaseDirectory": {
      "type": "boolean",
      "description": "Use pascal case directory name (e.g. `App/App.vue`).",
      "alias": "R",
      "default": false
    },
    "fileName": {
      "type": "string",
      "description": "Create a component with this file name."
    },
    "inSourceTests": {
      "type": "boolean",
      "default": false,
      "description": "When using Vitest, separate spec files will not be generated and instead will be included within the source files. Read more on the Vitest docs site: https://vitest.dev/guide/in-source.html"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "jest", "none"],
      "description": "Test runner to use for unit tests.",
      "x-prompt": "What unit test runner should be used?"
    }
  },
  "required": ["name", "project"]
}
