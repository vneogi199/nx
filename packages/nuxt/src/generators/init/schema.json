{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxNuxtInit",
  "title": "Init Nuxt Plugin",
  "description": "Initialize a Nuxt Plugin.",
  "cli": "nx",
  "type": "object",
  "properties": {
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "rootProject": {
      "description": "Create a project at the root of the workspace",
      "type": "boolean",
      "default": false
    },
    "skipPackageJson": {
      "description": "Do not add dependencies to `package.json`.",
      "type": "boolean",
      "default": false
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css"
    }
  },
  "required": []
}
