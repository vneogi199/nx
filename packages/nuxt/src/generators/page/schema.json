{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "NxNuxtPage",
  "title": "Create a Nuxt page",
  "description": "Create a Nuxt page for Nx.",
  "type": "object",
  "examples": [
    {
      "command": "nx g page new-page --project=myapp",
      "description": "Generate a page in the `myapp` application"
    }
  ],
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this page?",
      "x-priority": "important"
    },
    "name": {
      "type": "string",
      "description": "The name of the page.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the page?",
      "x-priority": "important"
    },
    "directory": {
      "type": "string",
      "description": "Create the page under this directory - all nested directories will be created under the pages directory.",
      "alias": "dir"
    },
    "fileName": {
      "type": "string",
      "description": "Create a component with this file name."
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    }
  },
  "required": ["name", "project"]
}
