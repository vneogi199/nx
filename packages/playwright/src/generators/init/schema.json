{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxPlaywrightInit",
  "title": "Playwright Init Generator",
  "description": "Initializes a Playwright project in the current workspace.",
  "type": "object",
  "properties": {
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description": "Do not add dependencies to `package.json`.",
      "x-priority": "internal"
    },
    "skipInstall": {
      "type": "boolean",
      "description": "Skip running `playwright install`. This is to ensure that playwright browsers are installed.",
      "default": false
    }
  },
  "required": []
}
