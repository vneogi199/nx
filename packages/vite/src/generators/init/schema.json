{
  "cli": "nx",
  "title": "Initialize Vite in the workspace.",
  "description": "Initialize Vite in the workspace.",
  "$id": "init-vite-plugin",
  "type": "object",
  "properties": {
    "uiFramework": {
      "type": "string",
      "description": "UI Framework to use for Vite.",
      "enum": ["react", "none"],
      "default": "react",
      "x-prompt": "What UI framework plugin should Vite use?"
    },
    "compiler": {
      "type": "string",
      "description": "Compiler to use for Vite when UI Framework is React.",
      "enum": ["babel", "swc"],
      "default": "babel"
    },
    "includeLib": {
      "type": "boolean",
      "description": "Add dependencies needed to build libraries.",
      "default": false
    },
    "testEnvironment": {
      "description": "The vitest environment to use. See https://vitest.dev/config/#environment.",
      "type": "string",
      "enum": ["node", "jsdom", "happy-dom", "edge-runtime"],
      "default": "jsdom"
    }
  },
  "examplesFile": "../../../docs/init-examples.md"
}
