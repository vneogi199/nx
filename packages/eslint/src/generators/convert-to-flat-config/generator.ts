import {
  formatFiles,
  getProjects,
  NxJsonConfiguration,
  ProjectConfiguration,
  readNxJson,
  Tree,
  updateJson,
  updateProjectConfiguration,
} from '@nx/devkit';
import { ConvertToFlatConfigGeneratorSchema } from './schema';
import { findEslintFile } from '../utils/eslint-file';
import { convertEslintJsonToFlatConfig } from './converters/json-converter';

export async function convertToFlatConfigGenerator(
  tree: Tree,
  options: ConvertToFlatConfigGeneratorSchema
) {
  const eslintFile = findEslintFile(tree);
  if (!eslintFile) {
    throw new Error('Could not find root eslint file');
  }
  if (!eslintFile.endsWith('.json')) {
    throw new Error(
      'Only json eslint config files are supported for conversion'
    );
  }

  // rename root eslint config to eslint.config.js
  convertRootToFlatConfig(tree, eslintFile);
  // rename and map files
  const projects = getProjects(tree);
  for (const [project, projectConfig] of projects) {
    convertProjectToFlatConfig(tree, project, projectConfig, readNxJson(tree));
  }
  // replace references in nx.json
  updateNxJsonConfig(tree);
  // install missing packages

  if (!options.skipFormat) {
    await formatFiles(tree);
  }
}

export default convertToFlatConfigGenerator;

function convertRootToFlatConfig(tree: Tree, eslintFile: string) {
  if (eslintFile.endsWith('.base.json')) {
    convertConfigToFlatConfig(
      tree,
      '',
      '.eslintrc.base.json',
      'eslint.base.config.js'
    );
  }
  convertConfigToFlatConfig(tree, '', '.eslintrc.json', 'eslint.config.js');
}

function convertProjectToFlatConfig(
  tree: Tree,
  project: string,
  projectConfig: ProjectConfiguration,
  nxJson: NxJsonConfiguration
) {
  if (tree.exists(`${projectConfig.root}/.eslintrc.json`)) {
    if (projectConfig.targets) {
      const eslintTargets = Object.keys(projectConfig.targets || {}).filter(
        (t) => projectConfig.targets[t].executor === '@nx/eslint:lint'
      );
      for (const target of eslintTargets) {
        // remove any obsolete `eslintConfig` options pointing to the old config file
        if (projectConfig.targets[target].options?.eslintConfig) {
          delete projectConfig.targets[target].options.eslintConfig;
        }
        updateProjectConfiguration(tree, project, projectConfig);
      }
      const nxHasLintTargets = Object.keys(nxJson.targetDefaults || {}).some(
        (t) =>
          (t === '@nx/eslint:lint' ||
            nxJson.targetDefaults[t].executor === '@nx/eslint:lint') &&
          projectConfig.targets?.[t]
      );
      if (nxHasLintTargets || eslintTargets.length > 0) {
        convertConfigToFlatConfig(
          tree,
          projectConfig.root,
          '.eslintrc.json',
          'eslint.config.js'
        );
      }
    }
  }
}

// update names of eslint files in nx.json
// and remove eslintignore
function updateNxJsonConfig(tree: Tree) {
  if (tree.exists('nx.json')) {
    updateJson(tree, 'nx.json', (json: NxJsonConfiguration) => {
      if (json.targetDefaults?.lint?.inputs) {
        const inputSet = new Set(json.targetDefaults.lint.inputs);
        inputSet.add('{workspaceRoot}/eslint.config.js');
        json.targetDefaults.lint.inputs = Array.from(inputSet);
      }
      if (json.targetDefaults?.['@nx/eslint:lint']?.inputs) {
        const inputSet = new Set(json.targetDefaults['@nx/eslint:lint'].inputs);
        inputSet.add('{workspaceRoot}/eslint.config.js');
        json.targetDefaults['@nx/eslint:lint'].inputs = Array.from(inputSet);
      }
      if (json.namedInputs?.production) {
        const inputSet = new Set(json.namedInputs.production);
        inputSet.add('!{projectRoot}/eslint.config.js');
        json.namedInputs.production = Array.from(inputSet);
      }
      return json;
    });
  }
}

function convertConfigToFlatConfig(
  tree: Tree,
  root: string,
  source: string,
  target: string
) {
  convertEslintJsonToFlatConfig(tree, root, source, target);
}
