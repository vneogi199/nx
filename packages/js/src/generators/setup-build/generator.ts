import {
  convertNxGenerator,
  ensurePackage,
  formatFiles,
  type GeneratorCallback,
  joinPathFragments,
  readProjectConfiguration,
  runTasksInSerial,
  type Tree,
  updateProjectConfiguration,
} from '@nx/devkit';
import { addSwcConfig } from '../../utils/swc/add-swc-config';
import { addSwcDependencies } from '../../utils/swc/add-swc-dependencies';
import { nxVersion } from '../../utils/versions';
import { SetupBuildGeneratorSchema } from './schema';

export async function setupBuildGenerator(
  tree: Tree,
  options: SetupBuildGeneratorSchema
): Promise<GeneratorCallback> {
  const tasks: GeneratorCallback[] = [];
  const project = readProjectConfiguration(tree, options.project);
  const buildTarget = options.buildTarget ?? 'build';

  project.targets ??= {};

  let mainFile: string;
  if (options.main) {
    mainFile = options.main;
  } else {
    const root = project.sourceRoot ?? project.root;
    for (const f of [
      joinPathFragments(root, 'main.ts'),
      joinPathFragments(root, 'main.js'),
      joinPathFragments(root, 'index.ts'),
      joinPathFragments(root, 'index.js'),
    ]) {
      if (tree.exists(f)) {
        mainFile = f;
        break;
      }
    }
  }
  if (!mainFile || !tree.exists(mainFile)) {
    throw new Error(
      `Cannot locate a main file for ${options.project}. Please specify one using --main=<file-path>.`
    );
  }

  let tsConfigFile: string;
  if (options.tsConfig) {
    tsConfigFile = options.tsConfig;
  } else {
    for (const f of [
      'tsconfig.lib.json',
      'tsconfig.app.json',
      'tsconfig.json',
    ]) {
      const candidate = joinPathFragments(project.root, f);
      if (tree.exists(candidate)) {
        tsConfigFile = candidate;
        break;
      }
    }
  }
  if (!tsConfigFile || !tree.exists(tsConfigFile)) {
    throw new Error(
      `Cannot locate a tsConfig file for ${options.project}. Please specify one using --tsConfig=<file-path>.`
    );
  }

  switch (options.bundler) {
    case 'vite': {
      const { viteConfigurationGenerator } = ensurePackage(
        '@nx/vite',
        nxVersion
      );
      const task = await viteConfigurationGenerator(tree, {
        buildTarget: options.buildTarget,
        project: options.project,
        newProject: true,
        uiFramework: 'none',
        includeVitest: false,
        includeLib: true,
        skipFormat: true,
      });
      tasks.push(task);
      break;
    }
    case 'esbuild': {
      const { esbuildProjectGenerator } = ensurePackage(
        '@nx/esbuild',
        nxVersion
      );
      const task = await esbuildProjectGenerator(tree, {
        main: mainFile,
        buildTarget: options.buildTarget,
        project: options.project,
        skipFormat: true,
      });
      tasks.push(task);
      break;
    }
    case 'rollup': {
      const { rollupProjectGenerator } = ensurePackage('@nx/rollup', nxVersion);
      const task = await rollupProjectGenerator(tree, {
        buildTarget: options.buildTarget,
        main: mainFile,
        project: options.project,
        skipFormat: true,
        compiler: 'tsc',
      });
      tasks.push(task);
      break;
    }
    case 'tsc': {
      const outputPath = joinPathFragments('dist', project.root);
      project.targets[buildTarget] = {
        executor: `@nx/js:tsc`,
        outputs: ['{options.outputPath}'],
        options: {
          outputPath,
          main: mainFile,
          tsConfig: tsConfigFile,
          assets: [],
        },
      };
      updateProjectConfiguration(tree, options.project, project);
      break;
    }
    case 'swc': {
      const outputPath = joinPathFragments('dist', project.root);
      project.targets[buildTarget] = {
        executor: `@nx/js:swc`,
        outputs: ['{options.outputPath}'],
        options: {
          outputPath,
          main: mainFile,
          tsConfig: tsConfigFile,
          assets: [],
        },
      };
      updateProjectConfiguration(tree, options.project, project);
      addSwcDependencies(tree);
      addSwcConfig(tree, project.root, 'es6');
    }
  }

  await formatFiles(tree);

  return runTasksInSerial(...tasks);
}

export default setupBuildGenerator;
export const setupBuildSchematic = convertNxGenerator(setupBuildGenerator);
