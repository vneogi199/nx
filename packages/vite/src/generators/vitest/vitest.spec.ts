import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import {
  Tree,
  readProjectConfiguration,
  updateProjectConfiguration,
} from '@nx/devkit';

import generator from './vitest-generator';
import { VitestGeneratorSchema } from './schema';
import {
  mockReactAppGenerator,
  mockReactLibNonBuildableJestTestRunnerGenerator,
} from '../../utils/test-utils';

describe('vitest generator', () => {
  let appTree: Tree;
  const options: VitestGeneratorSchema = {
    project: 'my-test-react-app',
    uiFramework: 'react',
    coverageProvider: 'c8',
  };

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('Should add the test target to existing test target', async () => {
    mockReactAppGenerator(appTree);
    await generator(appTree, options);
    const config = readProjectConfiguration(appTree, 'my-test-react-app');
    expect(config.targets['test']).toMatchInlineSnapshot(`
      {
        "executor": "@nx/vite:test",
        "options": {
          "passWithNoTests": true,
        },
        "outputs": [
          "{workspaceRoot}/coverage/{projectRoot}",
        ],
      }
    `);
  });

  it('should add the test target if its missing', async () => {
    mockReactAppGenerator(appTree);
    const projectConfig = readProjectConfiguration(
      appTree,
      'my-test-react-app'
    );
    delete projectConfig.targets.test;
    updateProjectConfiguration(appTree, 'my-test-react-app', projectConfig);
    await generator(appTree, options);
    const config = readProjectConfiguration(appTree, 'my-test-react-app');
    expect(config.targets['test']).toMatchInlineSnapshot(`
      {
        "executor": "@nx/vite:test",
        "options": {
          "passWithNoTests": true,
          "reportsDirectory": "../../coverage/apps/my-test-react-app",
        },
        "outputs": [
          "{options.reportsDirectory}",
        ],
      }
    `);
  });

  describe('tsconfig', () => {
    it('should add a tsconfig.spec.json file', async () => {
      mockReactAppGenerator(appTree);
      await generator(appTree, options);
      const tsconfig = JSON.parse(
        appTree.read('apps/my-test-react-app/tsconfig.json')?.toString() ?? '{}'
      );
      expect(tsconfig.references).toMatchInlineSnapshot(`
        [
          {
            "path": "./tsconfig.app.json",
          },
          {
            "path": "./tsconfig.spec.json",
          },
        ]
      `);

      const tsconfigSpec = JSON.parse(
        appTree.read('apps/my-test-react-app/tsconfig.spec.json')?.toString() ??
          '{}'
      );
      expect(tsconfigSpec).toMatchInlineSnapshot(`
        {
          "compilerOptions": {
            "outDir": "../../dist/out-tsc",
            "types": [
              "vitest/globals",
              "vitest/importMeta",
              "vite/client",
              "node",
              "vitest",
            ],
          },
          "extends": "./tsconfig.json",
          "include": [
            "vite.config.ts",
            "src/**/*.test.ts",
            "src/**/*.spec.ts",
            "src/**/*.test.tsx",
            "src/**/*.spec.tsx",
            "src/**/*.test.js",
            "src/**/*.spec.js",
            "src/**/*.test.jsx",
            "src/**/*.spec.jsx",
            "src/**/*.d.ts",
          ],
        }
      `);
    });

    it('should add vitest/importMeta when inSourceTests is true', async () => {
      mockReactAppGenerator(appTree);
      await generator(appTree, { ...options, inSourceTests: true });
      const tsconfig = JSON.parse(
        appTree.read('apps/my-test-react-app/tsconfig.app.json')?.toString() ??
          '{}'
      );
      expect(tsconfig.compilerOptions.types).toMatchInlineSnapshot(`
        [
          "vitest/importMeta",
        ]
      `);
    });
  });

  describe('vite.config', () => {
    it('should create correct vite.config.ts file for apps', async () => {
      mockReactAppGenerator(appTree);
      await generator(appTree, options);
      expect(
        appTree.read('apps/my-test-react-app/vite.config.ts', 'utf-8')
      ).toMatchSnapshot();
    });

    it('should create correct vite.config.ts file for non buildable libs', async () => {
      mockReactLibNonBuildableJestTestRunnerGenerator(appTree);
      await generator(appTree, { ...options, project: 'react-lib-nonb-jest' });
      expect(
        appTree.read('libs/react-lib-nonb-jest/vite.config.ts', 'utf-8')
      ).toMatchSnapshot();
    });
  });

  describe('insourceTests', () => {
    it('should add the insourceSource option in the vite config', async () => {
      mockReactAppGenerator(appTree);
      await generator(appTree, { ...options, inSourceTests: true });
      expect(
        appTree.read('apps/my-test-react-app/vite.config.ts', 'utf-8')
      ).toMatchSnapshot();
    });
  });
});
