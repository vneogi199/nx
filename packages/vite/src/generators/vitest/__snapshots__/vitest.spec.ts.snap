// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vitest generator insourceTests should add the insourceSource option in the vite config 1`] = `
"/// <reference types='vitest' />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/my-test-react-app',

  plugins: [react(), nxViteTsPaths()],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },

  define: {
    'import.meta.vitest': undefined,
  },
  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    includeSource: ['src/**/*.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;

exports[`vitest generator vite.config should create correct vite.config.ts file for apps 1`] = `
"/// <reference types='vitest' />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/my-test-react-app',

  plugins: [react(), nxViteTsPaths()],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },

  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;

exports[`vitest generator vite.config should create correct vite.config.ts file for non buildable libs 1`] = `
"/// <reference types='vitest' />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/react-lib-nonb-jest',

  plugins: [react(), nxViteTsPaths()],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },

  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;
